pkg/vdri/bloc/vdri.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package httpbinding
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	vdriapi "github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdri"
+	"github.com/hyperledger/aries-framework-go/pkg/vdri/httpbinding"
+	"github.com/sirupsen/logrus"
+	"github.com/trustbloc/sidetree-core-go/pkg/restapi/model"
+)
+
+var logger = logrus.New()
+
+// VDRI bloc
+type VDRI struct {
+	domain      string
+	resolverURL string
+}
+
+// createPayloadSchema is the struct for create payload
+type createPayloadSchema struct {
+
+	// operation
+	Operation model.OperationType `json:"type"`
+
+	// Encoded original DID document
+	DidDocument string `json:"didDocument"`
+
+	// Hash of the one-time password for the next update operation
+	NextUpdateOTPHash string `json:"nextUpdateOtpHash"`
+
+	// Hash of the one-time password for this recovery/checkpoint/revoke operation.
+	NextRecoveryOTPHash string `json:"nextRecoveryOtpHash"`
+}
+
+// New creates new bloc vdri
+func New(domain, resolverURL string) (*VDRI, error) {
+	// Validate host
+	_, err := url.ParseRequestURI(domain)
+	if err != nil {
+		return nil, fmt.Errorf("base URL invalid: %w", err)
+	}
+
+	// Validate host
+	_, err = url.ParseRequestURI(resolverURL)
+	if err != nil {
+		return nil, fmt.Errorf("base URL invalid: %w", err)
+	}
+
+	return &VDRI{domain: domain, resolverURL: resolverURL}, nil
+}
+
+// Accept did method
+func (v *VDRI) Accept(method string) bool {
+	return method == "bloc"
+}
+
+// Store did doc
+func (v *VDRI) Store(doc *did.Doc, by *[]vdriapi.ModifiedBy) error {
+	logger.Debugf("store not supported in bloc vdri")
+	return nil
+}
+
+// Build did doc
+func (v *VDRI) Build(pubKey *vdriapi.PubKey, opts ...vdriapi.DocOpts) (*did.Doc, error) {
+	sideTreePeersURL := discoverSideTreePeers(v.domain)
+
+	if len(sideTreePeersURL) == 0 {
+		return nil, errors.New("could not discover any side tree peers")
+	}
+
+	// TODO how to choose which peer
+	sideTreeVDRI, err := httpbinding.New(sideTreePeersURL[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new sidetree vdri: %w", err)
+	}
+
+	opts = append(opts, vdriapi.WithRequestBuilder(buildSideTreeRequest))
+
+	resDoc, err := sideTreeVDRI.Build(pubKey, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create did: %w", err)
+	}
+
+	return resDoc, nil
+}
+
+func (v *VDRI) Read(didID string, opts ...vdriapi.ResolveOpts) (*did.Doc, error) {
+	resolver, err := httpbinding.New(v.resolverURL)
```

Comment:
We're not resolving against a uniresolver. We're resolving against the peers we've discovered via `discoverSideTreePeers()`.

Reply:
Ignore this comment. After this morning's discussion it's clear we'll be resolving via the uniresolver (and passing policies).